### DISCUSSION TITLE
modified metrics for NER

### DISCUSSION THREAD

**Discussion by m-nlp-q** [explosion/spaCy#7494] [Help: Coding & Implementations] [answered]
Hello,

I am further training `el_core_news_lg` for the NER task. This component provides `ents_f` score to monitor the performance of the model. I would like to exclude the `O` label from the metric calculation and have a score for all other labels. Is there a way to achieve this from the training config file?

**Comment by adrianeboyd** [comment, marked as answer]
Hi, the `ents_f` score comes from this (undocumented) method, which calculates micro-PRF for entity spans as `(start, end, label)` units:

https://github.com/explosion/spaCy/blob/34e13c1161f7d42b961026b12d2eb3d3165bae27/spacy/scorer.py#L688-L730

The main complication is handling missing annotation in reference docs. Otherwise it's a standard PRF calculation.

**Reply by m-nlp-q** [reply]
@adrianeboyd thank you for your fast response.

So, if I am getting it right, the `ents_f` is calculating exactly what I described?

**Reply by adrianeboyd** [reply]
Well, "exclude O" isn't really the whole story, but the evaluation method looks at the set of entity spans, not the token-level IOB tags.

**Reply by m-nlp-q** [reply]
So the metrics are indeed calculated just on entities but on the span-level, where entity spans are provided by the training dataset.

I guess that the model assigns IOB tags in the token-level to the tokenized text that the tokenizer produces internally. Thus the metrics are calculated on the entity spans provided by the training dataset and the entity spans derived from the token tags assigned by the model. 

If this is the case, the `O` spans are ignored, which is exactly to what I am looking for.

@adrianeboyd you were very helpful.
